packages/embed/src/js/metadata/index.js

```diff
@@ -0,0 +1,240 @@
+import { getFingerprint } from './fingerprint';
+import { parseUserAgent } from './user-agent';
+
+const defaultMetadata = {
+  device_model: null, // string
+  device_os: null, // string
+  browser_name: null, // string
+  browser_version: null, // string
+  document_back_capture_duration_ms: null, // number
+  document_back_capture_retries: null, // number
+  document_back_image_origin: null, // uploaded | camera
+  document_back_capture_camera_name: null, // string
+  document_front_capture_duration_ms: null, // number
+  document_front_capture_retries: null, // number
+  document_front_image_origin: null, // uploaded | camera
+  document_front_capture_camera_name: null, // string
+  selfie_capture_duration_ms: null, // number
+  selfie_image_origin: null, // uploaded | camera
+  camera_name: null, // string - for selfies
+  active_liveness_type: null, // headpose | smile
+  active_liveness_version: null, // string
+  fingerprint: null, // string
+  user_agent: null, // string
+};
+
+let capturing = null;
+let activeCameraName = null;
+let metadata = { ...defaultMetadata };
+let captureStartTimestamp = null;
+
+/**
+ * Initialize the metadata object with some default values and data that
+ * can be collected from the browser. This function should be called once
+ * when the script is loaded.
+ *
+ * @returns {Promise<void>}
+ */
+export const initializeMetadata = async () => {
+  metadata = { ...defaultMetadata };
+  metadata.user_agent = navigator.userAgent;
+  const parsedUserAgent = parseUserAgent(navigator.userAgent);
+  metadata.device_model = parsedUserAgent.device || null;
+  metadata.device_os = parsedUserAgent.os;
+  metadata.browser_name = parsedUserAgent.browser;
+  metadata.browser_version = parsedUserAgent.browserVersion;
+  metadata.fingerprint = await getFingerprint();
+  metadata.active_liveness_type = 'smile';
+  metadata.active_liveness_version = '0.0.1';
```

Comment:
we will have to confirm what liveness version to use or if we can ignore this for now since we don't have any model